new package: x265

darealshinji opened the pull request.

TimothyGu commented on src/x265.mk:
```diff
@@ -0,0 +1,65 @@
+    $(MAKE) -C '$(1)/8bit' -j '$(JOBS)' install
+    $(INSTALL) '$(1)/8bit/libx265_main12.a' '$(PREFIX)/$(TARGET)/lib/libx265_main12.a'
+    $(INSTALL) '$(1)/8bit/libx265_main10.a' '$(PREFIX)/$(TARGET)/lib/libx265_main10.a'
+    $(SED) -i 's|-lx265|-lx265 -lx265_main10 -lx265_main12|' '$(PREFIX)/$(TARGET)/lib/pkgconfig/x265.pc'
```
No, the `-DEXTRA_LIB` line above already added the 10- and 12-bit symbols to the libx265.a archive.

darealshinji replied:
Are you sure? Afaik that only adds them to the CLI.

tonytheodore replied:
> Afaik that only adds them to the CLI.

That seems correct (checking with `nm`), all three need to be installed. Note that we don't build executables, just the libraries.

TimothyGu commented on src/x265.mk:
```diff
@@ -0,0 +1,65 @@
+    cd '$(1)' && mkdir -p 8bit 10bit 12bit
+
+    # 12 bit
+    test '$(TARGET)' = 'x86_64-w64-mingw32.static' && X265_ASM=ON || X265_ASM=OFF; \
```
Why? I believe x265's assembly optimization works on 32-bit as well.

darealshinji replied:
Unfortunately it doesn't work anymore.
https://bitbucket.org/multicoreware/x265/issues/150/linux-x86-32-bit-yasm-error

tonytheodore replied:
The HIGH_BIT_DEPTH options are not supported on 32-bit, do we really want to enable them?

http://hg.videolan.org/x265/file/f8b8ebdc5457/source/CMakeLists.txt#l302

darealshinji replied:
Disabling everything which is not fully supported on 32 bit builds sounds reasonable to me.

tonytheodore replied:
I've got a new `option(USE_HIGH_BIT_DEPTH_SUFFIX "Name libraries according to bit depth" OFF)` etc. working so we can just install them and let people decide if they want to use them.

tonytheodore commented on src/x265.mk:
```diff
@@ -0,0 +1,65 @@
+$(PKG)_SUBDIR   := x265_$($(PKG)_VERSION)
+$(PKG)_FILE     := x265_$($(PKG)_VERSION).tar.gz
+$(PKG)_URL      := https://ftp.videolan.org/pub/videolan/$(PKG)/$($(PKG)_FILE)
+$(PKG)_DEPS     := gcc cmake yasm
```
`cmake` is built automatically and shouldn't be listed in DEPS

tonytheodore commented:
I've got the shared builds (mostly) working and I'll use this as an example of how we separate the core mxe (libs only) with the relaxed "handy tools" available via plugins. Leave it with me.

darealshinji commented:
I've updated the files and added a test. Now shared libraries are build too. The library name however is `libx265.dll` without the API.
I'm not so sure on how to rebase this because of the master branch changes I've merged in. I could make a new PR if you really want only 1 commit.

tonytheodore commented on src/x265.mk:
```diff
@@ -0,0 +1,74 @@
+# See index.html for further information.
+
+PKG             := x265
+$(PKG)_IGNORE   :=
```
add WEBSITE and use single line `# This file is part of MXE. See LICENSE.md for licensing information.`

tonytheodore commented on src/x265.mk:
```diff
@@ -0,0 +1,74 @@
+        -DENABLE_SHARED=OFF \
+        -DENABLE_ASSEMBLY=$(if $(findstring x86_64,$(TARGET)),ON,OFF) \
+        -DENABLE_CLI=OFF \
+	    -DWINXP_SUPPORT=ON \
```
extra indent

darealshinji commented:
File updated.

tonytheodore commented:
Thanks, normally would prefer to avoid `sed` but there's not much that can be done in this case.
